solutions/usaco-945.mdx

```diff
@@ -2,16 +2,71 @@
 id: usaco-945
 source: USACO Gold 2019 US Open
 title: Snakes
-author: Sofia Yang
+author: Sofia Yang, Ryan Chou
 ---
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_snakes_gold_open19.html)
 
+
+Instead of keeping track of all of the different resizings, let's assume that $dp[i][j]$ stores the minimum sum of all net sizes for the first $i$ snakes after $j$ resizings. Then the total wasted space would be $dp[n][k] - T$, where $T$ is the sum of all of the groups of snakes. 
+
 ## Implementation
 
 **Time Complexity:** $\mathcal{O}(N^3)$
 
 <LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int MX_SZ = 400;
+
+int main() {
+	freopen("snakes.in", "r", stdin);
+	freopen("snakes.out", "w", stdout);
+	
+	int n, k;
+	cin >> n >> k;
+
+	vector<int> groups(n + 1);
+
+	for (int i = 1; i <= n; i++) {
+		cin >> groups[i];
+	}
+
+	// dp[i][j] = min sum of net sizes after catching first i snakes with j resizings
+	vector<vector<int>> dp(MX_SZ + 1, vector<int>(MX_SZ + 1, 0));
+
+	int lg = -1, spaceused = 0;
```

Comment:
what the heck is `lg`
`spaceused` i kinda get but it's still up in the air

Reply:
largest group (ill rename to make it more clear)